src/core/layertree/qgslayertreenode.h

```diff
@@ -227,11 +227,32 @@ class CORE_EXPORT QgsLayerTreeNode : public QObject
     QVariant customProperty( const QString &key, const QVariant &defaultValue = QVariant() ) const;
     //! Remove a custom property from layer. Properties are stored in a map and saved in project file.
     void removeCustomProperty( const QString &key );
+
+    // TODO: Harmonize this with QgsMapLayer::customPropertyKeys() for QGIS 4 API
```

Comment:
You could do that now -- add the new method and deprecate customProperties(). That'll save work when QGIS 4.0 time rolls around...

Reply:
Good, just a question: what would be preferred in the context of this PR?

In my local branch I've deprecated the method and while compiling I get a couple of warnings. Should I also point all existent calls to the new method?

I ask this because it will bring other parts of the source code (`qgslayoutlegendwidget.cpp` and `qgsmaplayerlegend.cpp`) into this PR. Is this something acceptable before releasing v3.16?